src/app/helpers/helperMethods.ts

```diff
@@ -1,37 +1,123 @@
 const path = require('path');
 const fs = require('fs');
+const hosts = [
+    "excel",
+    "onenote",
+    "outlook",
+    "powerpoint",
+    "project",
+    "word"
+];
 
 export namespace helperMethods {
-
-export function deleteFolderRecursively(projectFolder: string) 
-{
-    if(fs.existsSync(projectFolder))
+    export function deleteFolderRecursively(projectFolder: string) 
     {
-        fs.readdirSync(projectFolder).forEach(function(file,index){ 
-        var curPath = projectFolder + "/" + file; 
-        
-        if(fs.lstatSync(curPath).isDirectory())
+        if(fs.existsSync(projectFolder))
         {
-            deleteFolderRecursively(curPath);
+            fs.readdirSync(projectFolder).forEach(function(file,index){ 
+            var curPath = projectFolder + "/" + file; 
+            
+            if(fs.lstatSync(curPath).isDirectory())
+            {
+                deleteFolderRecursively(curPath);
+            }
+            else
+            {
+                fs.unlinkSync(curPath);
+            }
+        }); 
+        fs.rmdirSync(projectFolder); 
         }
-        else
+    }
+
+    export function doesProjectFolderExists(projectFolder: string)
+    {      
+    if (fs.existsSync(projectFolder))
         {
-            fs.unlinkSync(curPath);
+        if (fs.readdirSync(projectFolder).length > 0)
+        {          
+            return true;
+        }
         }
-    }); 
-    fs.rmdirSync(projectFolder); 
+        return false;
+    };
+
+    export function cleanupProjectFolder(projectFolder: string, host: string, typescript: boolean) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                await _updateMoveProjectFiles(projectFolder, host, typescript);
+                await _modifyPackageJsonFile(projectFolder, host);
+                return resolve();
+            } catch (err){
+                return reject(err);
+            }
+        });
     }
-};
 
-export function doesProjectFolderExists (projectFolder: string)
-{      
-  if (fs.existsSync(projectFolder))
-    {
-      if (fs.readdirSync(projectFolder).length > 0)
-      {          
-        return true;
-      }
+    function _updateMoveProjectFiles(projectFolder: string, host: string, typescript: boolean) {
+        return new Promise(async (resolve, reject) => {
+            // copy host-specific manifest over manifest.xml
+            fs.readFile(path.resolve(`${projectFolder}/manifest.${host}.xml`), 'utf8', (err, contents) => {
```

Comment:
It would be good to have this in a function that specifically is named for what it does, convert the project to a single host. I am not sure if _updateMoveProjectFiles() is a required name or can be renamed, e.g. convertProjectToSingleHost().

Reply:
I would find it simpler to use fs.promise methods so you can write using await rather than the callbacks with the promises. I don't think you need to rewrite it but it would simplify the error handling.

Reply:
I was also thinking that this logic should ideally go into the project so that different projects could have different code to run. This is not required now for the sake of doing it, but it would be better. This code would also need to be skipped for Excel-Custom-Functions[-JS] templates as they are already for a single host.

Reply:
Good suggestions. I wasn't aware of fs.promise - good to know!

Reply:
I see that when using fs.promises, a warning is displayed: "ExperimentalWarning: The fs.promises API is experimental"

Is there a way to suppress this warning?  It's not something we want users seeing when they run yo office